src/main/java/com/epam/controller/ToursController.java

```diff
@@ -102,64 +114,83 @@ public ModelAndView addTour(@RequestParam String tourType, @RequestParam String
         ModelAndView toursModel = new ModelAndView();
         List<Hotel> hotels = hotelService.getHotels();
         toursModel.addObject("hotelList", hotels );
+        toursModel.addObject("tourType", tourType);
+        toursModel.addObject("startDate", startDate);
+        toursModel.addObject("endDate", endDate);
+        toursModel.addObject("price", pricePerPerson);
+        toursModel.addObject("discount", discount);
+        toursModel.addObject("description", tourDescription);
+        toursModel.setViewName("addtour");
+        LocalDate addStartDate;
+        LocalDate addEndDate;
         try {
-            LocalDate addStartDate = Validator.getDate(startDate, false);
-            LocalDate addEndDate = Validator.getDate(endDate, false);
-            Integer addPricePerPerson = Validator.getInt(pricePerPerson);
-            Integer addDiscount = Validator.getDiscount(discount);
-            Validator.checkEmpty(tourType);
-            Validator.checkEmpty(tourDescription);
-            Validator.dateDifference(addStartDate,addEndDate);
+            if(!startDate.isEmpty()) {
+                addStartDate = Date.valueOf(startDate).toLocalDate();
```

Comment:
Please think about collapsing this code. I see several places with duplicate logic

Reply:
Done